packages/x-components/src/components/base-dropdown.vue

```diff
@@ -91,382 +96,312 @@
   /**
    * Dropdown component that mimics a Select element behavior, but with the option
    * to customize the toggle button and each item contents.
-   *
-   * @public
    */
-  @Component({
-    components: {
-      NoElement
+  export default defineComponent({
+    name: 'BaseDropdown',
+    components: { NoElement },
+    props: {
+      /** List of items to display.*/
+      items: {
+        type: Array as PropType<DropdownItem[]>,
+        required: true
+      },
+      /** The selected item. */
+      modelValue: {
+        type: null as unknown as PropType<DropdownItem | null>,
```

Comment:
I think @joseacabaneros explained it last week during your PTO. I think the reason behind this was that you can't set a prop as required and null at the same time. However, this type is not actually enforcing anything but you can achieve what you want through a validator. Probably he can share the link to the issue where he found this.

About setting this prop as optional... I don't know, might be an option. Not sure if it could bring unexpected consequences 🤔

Reply:
Yes! That's right! https://github.com/vuejs/core/issues/3948#issuecomment-860466204

<img width="577" alt="Screenshot 2024-04-22 at 6 49 21 PM" src="https://github.com/empathyco/x/assets/10746604/ba917123-178a-41e0-b8f9-7033f79324c4">

And about optional, I also prefer to keep it as it is for what may happen 😆